### PR TITLE
Stop pinning reform-rails to pick up patch in 0.2.5

### PR DESCRIPTION
**Pull request by mjgiarlo**
## Why was this change made? 🤔

Keep pace with reform-rails now that the bug we found was patched.

## How was this change tested? 🤨

CI

### CODE REVIEW

**Review by jmartin-sul — COMMENTED** [review]
it looks like this unpins, but doesn't actually bump the version?  i still see [0.2.3 in `Gemfile.lock`](https://github.com/sul-dlss/argo/blob/2ae0f1ca796eec1c7478993c995ebf47a4c891d6/Gemfile.lock#L452)?

**Comment by mjgiarlo** [comment]
@jmartin-sul 💬 
> it looks like this unpins, but doesn't actually bump the version? i still see [0.2.3 in `Gemfile.lock`](https://github.com/sul-dlss/argo/blob/2ae0f1ca796eec1c7478993c995ebf47a4c891d6/Gemfile.lock#L452)?

Good eye! I've got that included now.

**Comment by mjgiarlo** [comment]
@jmartin-sul Whaddya know? I'll take a look. Thanks for putting this through its paces.

**Comment by mjgiarlo** [comment]
Wrote up a related bug issue in reform-rails: https://github.com/trailblazer/reform-rails/issues/101 I'll

**Comment by mjgiarlo** [comment]
@jmartin-sul I made a change to the populator in the `TagsForm` that fixes the specs atop reform-rails 0.2.5 (and still works atop 0.2.3, FWIW). I tested the patch in QA by issuing a variety of tag changes (simple add, simple delete, simple update, combo of adds and updates and deletes), and all tests I performed had the expected results.

I also paired with @jcoyne to try and make sense of how this ever worked... and we were both stumped. Given the pattern Argo's `TagsForm` uses in its populator is not used *anywhere* else in our portfolio, I was not inclined to give grokking this any more time than I already have but check me on this.

Long story short, I think this is ready for another look.

**Review by aaron-collier — APPROVED** [review]
